iOS_SDK/OneSignalSDK/Source/OSDevice.m

```diff
@@ -30,36 +30,30 @@
 
 @implementation OSDevice
 
-- (BOOL)isNotificationEnabled {
-    return [[[OneSignal getPermissionSubscriptionState] permissionStatus] reachable];
-}
-
-- (BOOL)isUserSubscribed {
-    return [[[OneSignal getPermissionSubscriptionState] subscriptionStatus] userSubscriptionSetting];
-}
-
-- (BOOL)isSubscribed {
-    return [[[OneSignal getPermissionSubscriptionState] subscriptionStatus] subscribed];
-}
-
-- (OSNotificationPermission)getNotificationPermissionStatus {
-    return [[[OneSignal getPermissionSubscriptionState] permissionStatus] status];
-}
-
-- (NSString *)getUserId {
-    return [[[OneSignal getPermissionSubscriptionState] subscriptionStatus] userId];
-}
-
-- (NSString *)getPushToken {
-    return [[[OneSignal getPermissionSubscriptionState] subscriptionStatus] pushToken];
-}
-
-- (NSString *)getEmailUserId {
-    return [[[OneSignal getPermissionSubscriptionState] emailSubscriptionStatus] emailUserId];
-}
-
-- (NSString *)getEmailAddress {
-    return [[[OneSignal getPermissionSubscriptionState] emailSubscriptionStatus] emailAddress];
+- (id)init {
+    return [[OSDevice alloc] initWithSubscriptionState:[OneSignal getPermissionSubscriptionState]];
+}
+
+- (instancetype)initWithSubscriptionState:(OSPermissionSubscriptionState *)state {
+    self = [super init];
+    if (self) {
+        _notificationEnabled = [[state permissionStatus] reachable];
```

Comment:
maybe an Objetive C question, this will assign the value of reachable to the notificationEnabled property or will this act as a pointer and whenever notificationEnabled is called it  will call reachable getter?